packages/@react-aria/dnd/stories/dnd.stories.tsx

```diff
@@ -0,0 +1,400 @@
+  let ref = React.useRef();
+  let {dropProps, isDropTarget} = useDrop({
+    ref,
+    onDropEnter: action('onDropEnter'),
+    // onDropMove: action('onDropMove'),
+    onDropExit: action('onDropExit'),
+    onDropActivate: action('onDropActivate'),
+    onDrop: action('onDrop'),
+    getDropOperation(types, allowedOperations) {
+      return !type || types.has(type) ? allowedOperations[0] : 'cancel';
+    }
+  });
+
+  let {buttonProps} = useButton({elementType: 'div'}, ref);
+
+  return (
+    <div
+      {...mergeProps(dropProps, buttonProps)}
+      ref={ref}
+      className={classNames(dropzoneStyles, 'spectrum-Dropzone', {'is-dragged': isDropTarget})}>
+      Drop here
+    </div>
+  );
+}
+
+function DialogButton({children}) {
+  let [isOpen, setOpen] = React.useState(false);
+  let ref = React.useRef();
+  let {dropProps, isDropTarget} = useDrop({
+    ref: unwrapDOMRef(ref),
+    onDropActivate() {
+      setOpen(true);
+    }
+  });
+
+  return (
+    <DialogTrigger isDismissable isOpen={isOpen} onOpenChange={setOpen}>
+      <PressResponder {...dropProps} isPressed={isDropTarget}>
+        <ActionButton ref={ref}>Open dialog</ActionButton>
+      </PressResponder>
+      {children}
+    </DialogTrigger>
+  );
+}
+
+function DraggableCollectionExample() {
+  let list = useListData({
+    initialItems: [
+      {id: 'foo', type: 'folder', text: 'Foo'},
+      {id: 'bar', type: 'folder', text: 'Bar'},
+      {id: 'baz', type: 'item', text: 'Baz'}
+    ]
+  });
+
+  let onDragEnd = (e) => {
+    if (e.dropOperation === 'move') {
+      list.remove(...e.keys);
+    }
+  };
+
+  return (
+    <DraggableCollection items={list.items} selectedKeys={list.selectedKeys} onSelectionChange={list.setSelectedKeys} onDragEnd={onDragEnd}>
+      {item => (
+        <Item textValue={item.text}>
+          {item.type === 'folder' && <Folder size="S" />}
+          <span>{item.text}</span>
+        </Item>
+      )}
+    </DraggableCollection>
+  );
+}
+
+function DraggableCollection(props) {
+  let ref = React.useRef<HTMLDivElement>(null);
+  let state = useListState(props);
+  let gridState = useGridState({
+    selectionMode: 'multiple',
+    collection: new GridCollection({
+      columnCount: 1,
+      items: [...state.collection].map(item => ({
+        ...item,
+        childNodes: [{
+          key: `cell-${item.key}`,
+          type: 'cell',
+          index: 0,
+          value: null,
+          level: 0,
+          rendered: null,
+          textValue: item.textValue,
+          hasChildNodes: false,
+          childNodes: []
+        }]
+      }))
+    })
+  });
+
+  let provider = useProvider();
+  let dragState = useDraggableCollectionState({
+    collection: gridState.collection,
+    selectionManager: gridState.selectionManager,
+    getItems(keys) {
+      return [...keys].map(key => {
+        let item = gridState.collection.getItem(key);
+
+        return {
+          // @ts-ignore
+          types: ['text/plain', item.value.type],
```

Comment:
This line sometimes throws a runtime error in Draggable Grid examples when `item` is not defined: 

```
dnd.stories.tsx:296 Uncaught TypeError: Cannot read property 'value' of undefined
    at dnd.stories.tsx:296
    at Array.map (<anonymous>)
    at Object.getItems (dnd.stories.tsx:291)
    at Object.getItems (useDraggableCollectionState.ts:56)
    at Object.getItems (useDraggableItem.ts:34)
    at onPress (useDrag.ts:229)
    at triggerPressEnd (usePress.ts:174)
    at HTMLDocument.onKeyUp (usePress.ts:275)
```

Reply:
Hmm, how did you reproduce this?